hooks/boot/01-kafo-hook-extensions.rb

```diff
@@ -139,6 +139,14 @@ def execute_command(command, do_say, do_log)
   def remote_host?(hostname)
     !['localhost', '127.0.0.1', `hostname`.strip].include?(hostname)
   end
+
+  def needs_postgresql_upgrade?(old_version)
+    File.read('/var/lib/pgsql/data/PG_VERSION').chomp.to_i == old_version.to_i
```

Comment:
Can this be generic:
```suggestion
  def needs_postgresql_upgrade?(new_version)
    File.read('/var/lib/pgsql/data/PG_VERSION').chomp.to_i < new_version.to_i
```

Reply:
I've debated this, and the answer is "it depends" (https://www.todepond.com/wikiblogarden/better-computing/just/depends/)

postgresql-upgrade 13 can only upgrade from 12. So if we feed it anything else, it will go belly up.

Should we have anything else than 12 at this point? No, probably not.

Reply:
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/deploying_different_types_of_servers/using-databases#fast-upgrade-using-the-pg_upgrade-tool_migrating-to-a-rhel-8-version-of-postgresql has the supported upgrade paths